src/main/java/sparksoniq/jsoniq/compiler/JsoniqExpressionTreeVisitor.java

```diff
@@ -109,7 +107,6 @@
     private Expression currentExpression;
     private Expression otherCurrentExpression;
 
-    private PrimaryExpression currentPrimaryExpression;
```

Comment:
Nice !

Reply:
I anticipated that you would like this one!